### PR TITLE
chore: Update backward compatibility section in CONTRIBUTING.md

### LINKED ISSUE #13045: Update backward compatibility promise in CONTRIBUTING.md to reflect HIP 0004
In the [Semantic Versioning section of CONTRIBUTING.md](https://github.com/helm/helm/blob/main/CONTRIBUTING.md#semantic-versioning) document, the backward compatibility promise does not align to [HIP-0004](https://github.com/helm/community/blob/main/hips/hip-0004.md)

- This ought to say public functions MUST not be changed in the `pkg/` directory

> Go libraries inside of `pkg/` SHOULD remain backward compatible, though code inside of `cmd/` and `internal/` may be changed from release to release without notice.
- SHOULD ought to be changed to MUST
- Question: Since `cmd/` is meant to be internal code, should have all code that isn't meant to be a subcommand live in `cmd/helm/internal`, to allow ease of sharing code within the `cmd/helm` package, but ensure no one can import that code?

The section should point to HIP-0004

- Should this also include changing functionality? If a public function/method should probably not change overall internal functionality e.g if a change removes previous functionality and makes the function a no-op, perhaps this should not be allowed.

### PR DESCRIPTION
**Pull request by banjoh**
**What this PR does / why we need it**:

Update backward compatibility section in CONTRIBUTING.md to reflect what [HIP-0004](https://github.com/helm/community/blob/main/hips/hip-0004.md) says

Fixes: https://github.com/helm/helm/issues/13045

**Special notes for your reviewer**:

**If applicable**:
- [ ] this PR contains documentation
- [ ] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

**Review by robertsirc — APPROVED** [review]
LGTM: this is just text updates on why libraries in pkg/ "MUST" remain backwards compatible.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -115,8 +115,9 @@ Helm maintains a strong commitment to backward compatibility. All of our changes
 formats are backward compatible from one major release to the next. No features, flags, or commands
 are removed or substantially modified (unless we need to fix a security issue).
 
-We also try very hard to not change publicly accessible Go library definitions inside of the `pkg/`
-directory of our source code.
+We also remain committed to not changing publicly accessible Go library definitions inside of the `pkg/` directory of our source code in a non-backwards-compatible way. For more on
```
Comment (scottrigby) [inline CONTRIBUTING.md:118]: just remove trailing duplicate "for more on"

```suggestion
We also remain committed to not changing publicly accessible Go library definitions inside of the `pkg/` directory of our source code in a non-backwards-compatible way.
```

**Review by scottrigby — CHANGES_REQUESTED** [review]
thanks for taking this on. agree, all looks good except one accidental half sentence. just clean that up and this will be good to go 🙂

**Comment by banjoh** [comment]
> thanks for taking this on. agree, all looks good except one accidental half sentence. just clean that up and this will be good to go 🙂

I've cleaned it up

**Review by scottrigby — APPROVED** [review]

**Comment by scottrigby** [comment]
Looks good! Thanks @banjoh

**Pull request merged**